chain-extensions/xvm/src/lib.rs

```diff
@@ -68,7 +68,7 @@ where
             XvmFuncId::XvmCall => {
                 // We need to immediately charge for the worst case scenario. Gas equals Weight in pallet-contracts context.
                 let remaining_weight = env.ext().gas_meter().gas_left();
-                let charged_weight = env.charge_weight(remaining_weight)?;
+                let charged_weight = env.charge_weight(remaining_weight.set_proof_size(0))?;
```

Comment:
Please add a comment (and a TODO) explaining why this is done and why it should be reverted later.

Also, I don't think you should set it to 0 - use some predefined dummy constant instead, something closer to real world scenarios (e.g. 32 kB).

Reply:
Updated. Please, note I rebased it over polkadot-v0.9.39 and changed branch.